code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -293,6 +293,9 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 	if(!sharer)
 		return 0
 
+	archive()
+	sharer.archive()

```

Comment:
Yeah, atmos is the most performant area of code. You need to find a way to do this in the specific use case